lale/operators.py

```diff
@@ -1794,6 +1795,42 @@ def customize_schema(
             self, schemas, relevantToOptimizer, constraint, tags, **kwargs
         )
 
+    def _propose_fixed_hyperparams(
+        self, key_candidates, hp_all, hp_schema, max_depth=2
+    ):
+        defaults = self.get_defaults()
+        explicit_defaults: Dict[str, Any] = {k: defaults[k] for k in key_candidates}
+
+        found: bool = False
+
+        for depth in range(0, max_depth):
+            if found:
+                return None
+            candidate_replacements: Any = list(
+                itertools.combinations(explicit_defaults.items(), depth + 1)
+            )
+            for replacements in candidate_replacements:
+                new_values = dict(replacements)
+                fixed_hp = {**hp_all, **new_values}
+                try:
+                    lale.type_checking.validate_schema(fixed_hp, hp_schema)
+                    found = True
+                    yield new_values
+                except jsonschema.ValidationError:
+                    pass
+
+    MAX_FIX_DEPTH: int = 2
+    MAX_FIX_SUGGESTIONS: int = 3
+
+    def _format_fix_suggestion(self, d: Dict[str, Any]) -> str:
```

Comment:
I suggest to call lale.pretty_print.hyperparams_to_string here: https://github.com/IBM/lale/blob/master/lale/pretty_print.py#L80

Reply:
great idea, thanks!